opensearch/src/test/java/org/opensearch/sql/opensearch/executor/OpenSearchExecutionEngineTest.java

```diff
@@ -252,4 +282,45 @@ public String explain() {
       return "explain";
     }
   }
+
+  public static class FakeOpenSearchIndexScan extends OpenSearchIndexScan {
+    @Getter
+    private String rawResponse;
+
+    public FakeOpenSearchIndexScan(OpenSearchClient client, Settings settings,
+                               String indexName, Integer maxResultWindow,
+                               OpenSearchExprValueFactory exprValueFactory,
+                                   String rawResponse) {
+      super(client, settings,
+          indexName,maxResultWindow, exprValueFactory);
+      this.rawResponse = rawResponse;
+    }
+
+    @Override
+    public void open() {
```

Comment:
To avoid making member variables protected why not leave this function in the base class and override fetchNextBatch instead?

Reply:
Changed in https://github.com/Bit-Quill/opensearch-project-sql/pull/217/commits/3de83d9c0a4932667b2078769d2a99c5167ea2fa. Thanks for the idea!